zone/groups.cpp

```diff
@@ -1126,7 +1126,12 @@ void Group::TeleportGroup(Mob* sender, uint32 zoneID, uint16 instance_id, float
 }
 
 bool Group::LearnMembers() {
-	std::string query = StringFormat("SELECT name FROM group_id WHERE groupid = %lu", (unsigned long)GetID());
+	std::string query = StringFormat("SELECT name FROM group_id "
+		"WHERE group_id.groupid = %lu AND group_id.name NOT "
```

Comment:
What is this trying to do

Reply:
I am unclear as to why the group system is broken into two tables, one for leaders specifically.

Reply:
Groups are made up of two variables for how they know a player is a member of the group.

- Mob* member
- char membername

When a player zones, member for that player needs to be set to nullptr. However as the player still needs to be a part of the group across zones, membername stays.

When a player zones Group::LearnMembers is used as part of the Group constructor to pull existing group information for that player to load the membername and Mob* member variables.

This is existing functionality, However I don't believe the secondary queries are required... as it appears in locations that LearnMembers is used, we are also doing a query for the group leader, and setting it as well. I will confirm

Reply:
Confirmed, the additional query isn't required. I've removed it... and cleaned up the original to use repositories.